GVFS/GVFS.Common/Maintenance/LooseObjectsStep.cs

```diff
@@ -21,27 +24,88 @@ public LooseObjectsStep(GVFSContext context, bool requireCacheLock, bool forceRu
         }
 
         public override string Area => nameof(LooseObjectsStep);
+        public int LooseObjectsPutIntoPackFile { get; private set; } = 0;
+        
+        // 50,000 was found to be the optimal time taking ~5 minutes
+        public int MaxLooseObjectsInPack { get; set; } = 50000;
 
         protected override string LastRunTimeFilePath => Path.Combine(this.Context.Enlistment.GitObjectsRoot, "info", LooseObjectsLastRunFileName);
-        protected override TimeSpan TimeBetweenRuns => TimeSpan.FromDays(7);
+        protected override TimeSpan TimeBetweenRuns => TimeSpan.FromDays(1);
 
         public int CountLooseObjects()
         {
             int count = 0;
-            
+
             foreach (string directoryPath in this.Context.FileSystem.EnumerateDirectories(this.Context.Enlistment.GitObjectsRoot))
             {
                 string directoryName = directoryPath.TrimEnd(Path.DirectorySeparatorChar).Split(Path.DirectorySeparatorChar).Last();
 
                 if (GitObjects.IsLooseObjectsDirectory(directoryName))
-                {                    
+                {
                     count += this.Context.FileSystem.GetFiles(Path.Combine(this.Context.Enlistment.GitObjectsRoot, directoryPath), "*").Count();
                 }
             }
 
             return count;
         }
 
+        public void PackLooseObjects(StreamWriter streamWriter)

```

Comment:
Minor naming comment:  This method does not actually pack loose objects.  Perhaps `WriteLooseObjectSHAs` would be a more accurate description? (I came across this while writing up the changes in https://github.com/wilbaker/VFSForGit/commit/3405d9bd55786ae861a0004fb4b05ece017cf71e).